packages/application-generic/src/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -265,4 +238,122 @@ export class TriggerEvent {
 
     return integration?.providerId as ProvidersIdEnum;
   }
+
+  private async triggerBroadcast(
+    command: TriggerEventCommand,
+    actorProcessed: SubscriberEntity,
+    template: NotificationTemplateEntity
+  ) {
+    const subscriberFetchBatchSize = 500;
+    let subscribers: SubscriberEntity[] = [];
+
+    for await (const subscriber of this.subscriberRepository.findBatch(
+      {
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+      },
+      'subscriberId',
+      {},
+      subscriberFetchBatchSize
+    )) {
+      subscribers.push(subscriber);
+      if (subscribers.length === subscriberFetchBatchSize) {
+        await this.sendToProcessSubscriberService(
+          command,
+          actorProcessed,
+          template,
+          subscribers
+        );
+        subscribers = [];
+      }
+    }
+
+    if (subscribers.length > 0) {
+      await this.sendToProcessSubscriberService(
+        command,
+        actorProcessed,
+        template,
+        subscribers
+      );
+    }
+  }
+
+  private async triggerMulticast(
+    command: TriggerEventCommand,
+    actorProcessed: SubscriberEntity,
+    template: NotificationTemplateEntity
+  ) {
+    const mappedRecipients = await this.mapTriggerRecipients.execute(
+      MapTriggerRecipientsCommand.create({
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        recipients: command.to,
+        transactionId: command.transactionId,
+        userId: command.userId,
+        actor: actorProcessed,
```

Comment:
Noticed that in the previous state is was using `mappedActor` Does anyone knows if there were particular reason? if there is i will refactor in order to pass `mappedActor`

Reply:
I personally have no idea.
@ainouzgali tagging you because it is my memory about the implementation of mapped actors was made by you, but I might be wrong, so apologies in advance.

Reply:
I believe only the subscriberId is needed, so thats why only the mapping was sent. But the command MapTriggerRecipientsCommand is expecting ISubscriberDefine, not SubscriberEntity.

Reply:
@djabarovgeorge , a few days ago my PR for adding actor was merged, let me know if you need help resolving conflicts or reintegrating it. As it seems we touched a lot of the same logic

Reply:
Thanks for bringing it up, i removed all of the actor/tenant i saw before the trigger event use-case (there was one redundant due to duplication in the flow), so now the trigger event use-case is responsible for the mapping.